akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -215,13 +239,19 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
           ex match {
             // every IllegalHttp2StreamIdException will be a GOAWAY with PROTOCOL_ERROR
             case e: Http2Compliance.IllegalHttp2StreamIdException ⇒
-              pushGOAWAY()
+              pushGOAWAY(debug = e.getMessage)
 
             case e: Http2Compliance.HeaderDecompressionFailed ⇒
-              pushGOAWAY(COMPRESSION_ERROR)
+              pushGOAWAY(COMPRESSION_ERROR, e.getMessage)
 
             case e: Http2Compliance.IllegalHttp2FrameSize ⇒
-              pushGOAWAY(FRAME_SIZE_ERROR)
+              pushGOAWAY(FRAME_SIZE_ERROR, e.getMessage)
```

Comment:
This looks as if it would benefit from the approach discussed above with removing all those specialized exception types.